Fix MySQL support and re-enable for CI

fritzm opened the pull request:
It looks like MySQL builds had been broken for a bit.  Decided to port forward to MySQL 8.0 while fixing, since it's been out since April 2018, and this was a bit of a pain.

MySQL 8.0 defaults to very conservative permissions wrt. `OUTFILE` and `LOCAL INFILE`; I guess that's a good thing but it took some extra shenanigans with the github action and the htm test sequence itself to work around this (and debugging github actions is itself pretty trying.)

In any case, CI builds of docs, MariaDB and MySQL are working with this, and it should help keep things from silently breaking again as we go forward...

fritzm committed:
Address MySQL 8 / MariaDB 10 API type differences

fritzm committed:
undeploy.sh MySQL 8.0 portability fixes

- Use information_schema.routines instead of mysql.proc for dropProc()

fritzm committed:
docs tests fixes for MySQL 8.0

- add stdout.flush() to docs test runner to ease debugging
- add "chmod" step to htm index example to account for MySQL 8.0
  writing OUTFILE with restricive permissions

fritzm committed:
mysqlversion.py fixes for MySQL 8.0

fritzm committed:
CI github action fixes for MySQL

- Reenable MySQL job
- Add step to adjust MySQL config to allow OUTFILE and LOCAL INFILE
- Fix mariadb/mysql copypasta

fritzm committed:
Moved TODOs to github issues

fritzm committed:
Added RELEASE-NOTES.md

smonkewitz commented on tools/docs.py:
```diff
@@ -439,8 +439,9 @@ def _test(obj):
             except subprocess.CalledProcessError:
                 print("Failed to run documentation example:\n\n%s\n\n" % ex.source,
                       file=sys.stderr)
+                sys.stdout.flush()
```
Should this be `sys.stderr`?

fritzm replied:
Actually stdout on purpose here, to flush the preceding "print" statement so error report outputs show up sequenced with other ongoing output instead of (confusingly) some time later.

smonkewitz replied:
The preceding print is to stderr though. If the issue is confusing interleaving of stdout and stderr, shouldn’t the stdout flush happen before the print? (And then you might want an additional stderr flush after, depending on whether Python line buffers or block buffers sys.stderr.) Alternatively, I think you could avoid confusing interleavings by running the interpreter with the `-u` flag.

smonkewitz commented on tools/mysqlversion.py:
```diff
@@ -39,12 +39,12 @@
     _MYSQL: {
-        _MIN_VERSION: (5, 0),
-        _EXACT_VERSION: ['5.1.65', '5.1.73']
+        _MIN_VERSION: (8, 0),
```
I think the minimum MySQL version mentioned by `README.md` and `tools/templates/sections.xml` needs to be updated as well.

smonkewitz replied:
And we should probably mention that MariaDB works too.

fritzm replied:
Good catch -- will do on both of these, thanks!

fritzm committed:
Update README and HTML docs

- Rationalize READM and HTML docs build/install/deploy section to same content
- Deployment script described in both places now; references to older make install removed
- Speak uniformly of MySQL/MariaDB throughout
- Update prerequisites (MySQL 8.x -or- MariaDB 10.x, mysqlclient instead of MySQLdb)

smonkewitz commented on README.md:
```diff
@@ -46,52 +46,69 @@ database prior to installing sciSQL. If you do not, YOU WILL LOSE DATA.
+    export PATH="$PREFIX/bin:$PATH"
+    export PYTHONPATH="$PREFIX/python:$PATH"
+
+Check that your have access to a local server instance and run `scisql-deploy.py`. Passing
```
s/your/you/

fritzm replied:
will fix, thx!

smonkewitz commented on README.md:
```diff
@@ -46,52 +46,69 @@ database prior to installing sciSQL. If you do not, YOU WILL LOSE DATA.
+use standard input, for example via a pipe, for providing this password.
+Connection details, including this password, are stored in a temporary directory in a file named
+`my-client.cnf` using the MySQL options file format. This temporary file is removed at the end
+of the process, unless you set the verbosity level to DEBUG
```
Missing period at the end of the sentence.

fritzm replied:
will fix, thx!

smonkewitz reviewed (approved):
I noticed a couple minor things in the README update, and the flush should perhaps be removed or changed, but generally I think this looks good. Thanks for the MySQL compatibility fixes!

fritzm committed:
Review feedback

- Remove misconceived `stdout.flush()` in docs.py.  Invoke python
  interpreter with -u option in CI github action instead
